src/runtimes/node/index.js

```diff
@@ -35,7 +50,8 @@ const zipFunction = async function ({
   srcPath,
   stat,
 }) {
-  const bundler = config.nodeBundler || getDefaultBundler({ extension })
+  // how do you construct the srcFile?
+  const bundler = config.nodeBundler || getDefaultBundler({ extension, srcFile: join(srcPath, filename) })
```

Comment:
`getDefaultBundler` returns a Promise, so depending on which branch of we fall into, `bundler` will be either `string` or `Promise<string>`, which may be problematic. How do you feel about:

```js
const bundler = config.nodeBundler || (await getDefaultBundler({ extension, srcFile: join(srcPath, filename) }))
```

Reply:
Also, I believe you can use `mainFile` instead of `join(srcPath, filename)`.

Reply:
I had tried mainFile but I kept getting results where it looked like mainFile wasn't the actual file being investigated? is it possible `getDefaultBundler` will be called repeatedly for the same mainFile but with different dependencies its evaluating?